test/com/google/enterprise/adaptor/filenet/DocumentTraverserTest.java

```diff
@@ -386,6 +387,36 @@ public void testGetDocContent_notModified() throws Exception {
     assertEquals(0, baos.size());
   }
 
+  @Test
+  public void testGetDocContent_notFound() throws Exception {
+    Id id = new Id("{AAAAAAAA-0000-0000-0000-000000000000}");
+    DocId docId = newDocId(id);
+    MockObjectStore os = getObjectStore();
+    mockDocument(os, id.toString(), DOCUMENT_TIMESTAMP, true);
+
+    DocumentTraverser traverser = new DocumentTraverser(options);
+    {
+      Request request = new MockRequest(docId);
+      ByteArrayOutputStream baos = new ByteArrayOutputStream();
+      RecordingResponse response = new RecordingResponse(baos);
+      traverser.getDocContent(id, request, response);
+
+      byte[] actualContent = baos.toByteArray();
+      assertEquals("sample content", new String(actualContent, UTF_8));
+    }
+
+    // Now delete the document.
+    os.deleteObject(id);
```

Comment:
What is gained by adding the document to the store, fetching, deleting it, and then fetching it again, compared to just never creating a document in the first place and trying to fetch it?